src/value.rs

```diff
@@ -156,26 +163,36 @@ impl Value {
         }
     }
 
-    /// Tries to get a value of type `T`.
-    ///
-    /// Returns `Some` if the type is correct and the value is not `None`.
+    /// Tries to get a possibly optional value of type `T`.
     ///
-    /// This function doesn't distinguish between type mismatches and correctly
-    /// typed `None` values. Use `downcast` or `is` for that.
-    pub fn get<'a, T: FromValueOptional<'a>>(&'a self) -> Option<T> {
+    /// Returns `Ok` if the type is correct.
+    pub fn get<'a, T: FromValueOptional<'a>>(&'a self) -> Result<Option<T>, BoolError> {
```

Comment:
This should probably be a new error type, some kind of
```rust
struct ValueTypeMismatchError;
```

Reply:
Yes I agree, but on the other hand, `BoolError` is used elsewhere, like in `{g,s}et_property`.

Reply:
No strong opinion, what you think is best here :) Or maybe @GuillaumeGomez has an opinion

I'm not a fan of `BoolError` for things where we explicitly handle different error cases (other than just a C function returning `false`), and I'd like to change things to explicit error types. Just didn't get to it, or even to create an issue for discussion :)

Reply:
I agree with @sdroege on this one. I prefer to have things as explicit as possible.